crates/matrix-sdk/src/matrix_auth/login_builder.rs

```diff
@@ -183,6 +185,22 @@ impl LoginBuilder {
         let response = client.send(request, Some(RequestConfig::short_retry())).await?;
         self.auth.receive_login_response(&response).await?;
 
+        // This may block login for a while, but the user asked for it!
```

Comment:
Eh, I don't think that this will be good enough. Maybe not something for this PR but I think we'll have to push this into a background task, otherwise some of our power users might complain that logging in takes too long.

Let's re-add the TODO from before.

Reply:
Done, opened an issue too so I can link from it.